### PR TITLE
Fix ESC to work at unit menu

### LINKED ISSUE #656: CSSValueInput doesn't work as expected with esc key
As of now 1st ESC press do nothing, 2nd reset the field to an empty value.
Should be:
1. If the menu is open - close the menu
2. If the menu is not open reset to the initial control value

### PR DESCRIPTION
**Pull request by istarkov**
closes #656

Following test case at #673 didn't work:

Click unit px
Hover over items
Press ESC - instead of closing menu it resets input and only 2nd ESC closes the menu.

*The issue is Tooltip + UnitSelect.*

UnitSelect set 'focus" on Item on hover.
So hovering over the menu causes a lot of blur/focus events dispatched to the ancestors.
Tooltip at the top closes/opens very fast (not visible but internally it is). 
That cause https://github.com/radix-ui/primitives/blob/e861c16f20fe8e2e440983355edf2ec78cc2a51c/packages/react/dismissable-layer/src/DismissableLayer.tsx#L105 to think that Tooltip popover is now at the top layer. And prevent `onEscapeKeyDown` to be executed.

Current hack somehow improves the situation. But seems like better would be to disable the tooltip in case of focus/blur of the whole control. Not sure it's needed at all as we have the label.

## Description

1. What is this PR about (link the issue and add a short description)

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [x] made a self-review
- [x] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [x] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/designer/features/style-panel/shared/css-value-input/unit-select.tsx [hunk 1]
```diff
@@ -112,6 +112,10 @@ const StyledTrigger = styled(TextFieldIconButton, textStyles, {
   px: 3,
 });
 
+const handlePreventEvent = (event: React.FocusEvent<HTMLDivElement>) => {
```
Comment (kof) [inline apps/designer/app/designer/features/style-panel/shared/css-value-input/unit-select.tsx:115]:
```suggestion
const handlePreventDefault = (event: React.FocusEvent<HTMLDivElement>) => {
```

**Review by TrySound — APPROVED** [review]

**Review by kof — APPROVED** [review]
if possible check if we can throw out react-use useDebounce so we have a single thing to learn

File: apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx [hunk 2]
```diff
@@ -29,10 +29,10 @@ export const TextControl = ({
   >();
 
   return (
-    <Tooltip
+    <EnhancedTooltip
```
Comment (kof) [inline apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx:32]: actually we need to use it for all controls, not only text-control

File: packages/design-system/src/components/tooltip.tsx [hunk 3]
```diff
@@ -85,6 +99,102 @@ export const Tooltip = React.forwardRef(function TooltipWrapper(
...
+
+    const onOpenChange = (isOpen: boolean) => {
+      if (allowOpenRef.current) {
+        setOpen(isOpen);
+      }
+    };
+
+    // Wrap in provider so each control will use own hover delay
+    return (
+      <TooltipPrimitive.Provider>
```
Comment (kof) [inline packages/design-system/src/components/tooltip.tsx:186]: can't we have a provider used separately wrapped around entire style panel? so we can set the settings for timeout on it in a single place?

File: packages/design-system/src/components/tooltip.tsx [hunk 4]
```diff
@@ -85,6 +99,102 @@ export const Tooltip = React.forwardRef(function TooltipWrapper(
   );
 });
 
+/**
+ * EnhancedTooltip has the following differences from the radix-ui
+ * 1. Don't show the tooltip if any click or key-down was made inside Tooltip.Trigger
+ * 2. Show Tooltip on focus after the delay
+ **/
+export const EnhancedTooltip = React.forwardRef(
+  function TextControlTooltipWrapper(
```
Comment (kof) [inline packages/design-system/src/components/tooltip.tsx:108]: doesn't really need a name

File: apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx [hunk 5]
```diff
@@ -29,10 +29,10 @@ export const TextControl = ({
...
       content={styleConfig.label}
-      delayDuration={600}
-      disableHoverableContent={true}
+      delayDuration={1600}
+      disableHoverableContent={false}
```
Comment (kof) [inline apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx:35]: I think you don't want to set this up for each tooltip individually

File: packages/design-system/src/components/tooltip.tsx (hunk 4 above)
Comment (istarkov) [reply packages/design-system/src/components/tooltip.tsx:108]: eslint display name rule, changed on Enhanced...

File: apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx (hunk 5 above)
Comment (istarkov) [reply apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx:35]: Yep. If `skipDelayDuration = 0` would work as expected.

File: packages/design-system/src/components/tooltip.tsx (hunk 4 above)
Comment (kof) [reply packages/design-system/src/components/tooltip.tsx]: The inner function will not help react dev tools afaik, you still need displayName in the end and everywhere else linter was fine

**Review by kof — APPROVED** [review]

**Pull request merged**